core/platform/datastore/cloud_datastore_services_test.py

```diff
@@ -0,0 +1,181 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for the cloud_datastore_services.py"""
+
+from __future__ import annotations
+
+import datetime
+
+from core import feconf
+from core.platform import models
+from core.platform.datastore import cloud_datastore_services
+from core.tests import test_utils
+
+from typing import Sequence, Tuple
+
+MYPY = False
+if MYPY:
+    from mypy_imports import datastore_services
+    from mypy_imports import user_models
+
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+datastore_services = models.Registry.import_datastore_services()
+
+
+class CloudDatastoreServicesTests(test_utils.GenericTestBase):
+    """Unit tests for the cloud_datastore_services.py"""
+
+    THREE_WEEKS = datetime.timedelta(weeks=3)
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
+        self.admin_id = self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL)
+        self.set_curriculum_admins([self.CURRICULUM_ADMIN_USERNAME])
+
+        self.login(self.CURRICULUM_ADMIN_EMAIL, is_super_admin=True)
+        self.admin_user_id = self.get_user_id_from_email(
+            self.CURRICULUM_ADMIN_EMAIL)
+        self.curr_time = datetime.datetime.utcnow()
+        self.completed_activities_model = user_models.CompletedActivitiesModel(
+            id=self.admin_user_id,
+            exploration_ids=[],
+            collection_ids=[],
+            story_ids=[],
+            learnt_topic_ids=[],
+            last_updated=self.curr_time
+        )
+        self.user_query_model = user_models.UserQueryModel(
+            id='query_id',
+            user_ids=[],
+            submitter_id=self.admin_user_id,
+            query_status=feconf.USER_QUERY_STATUS_PROCESSING,
+            last_updated=self.curr_time
+        )
+
+    def test_update_timestamps_multi(self) -> None:
+        cloud_datastore_services.update_timestamps_multi(
+            [self.completed_activities_model, self.user_query_model], False)
+        cloud_datastore_services.put_multi(
+            [self.completed_activities_model, self.user_query_model])
+
+        self.assertEqual(
+            self.completed_activities_model.get_by_id(
+                self.admin_user_id).last_updated,
+            self.curr_time)
+        self.assertEqual(
+            self.user_query_model.get_by_id('query_id').last_updated,
+            self.curr_time)
```

Comment:
ditto: please add pre-checks here and elsewhere

Reply:
Done